src/AVFoundation/AVAudioPlayer.cs

```diff
@@ -38,24 +38,34 @@ public partial class AVAudioPlayer {
 				IntPtr errhandle;
 				IntPtr ptrtohandle = (IntPtr) (&errhandle);
 
-				var ap = new AVAudioPlayer (url, ptrtohandle);
```

Comment:
In general I think it's better to avoid the exception in the first place, instead of handling an expected exception.

Unfortunately this would mean implementing the binding manually, which would be something like this:

```cs
public static AVAudioPlayer? FromUrl (NSUrl url, out NSError? error)
{
	error = null;

	var handle = Messaging.IntPtr_objc_msgSend (class_ptr, Selector.GetHandle ("alloc"));
	if (handle == IntPtr.Zero)
		return null;
	var errorptr = IntPtr.Zero;
	unsafe {
		handle = objc_msgSend (handle, Selector.GetHandle ("initWithData:error:"), &errorptr);
	}
	error = Runtime.GetNSObject<NSError> (errorptr);
	return Runtime.GetNSObject<AVAudioPlayer> (handle, owns: true);
}

[DllImport (Constants.ObjectiveCLibrary, EntryPoint = "objc_msgSend")]
unsafe static extern IntPtr objc_msgSend (IntPtr receiver, IntPtr selector, IntPtr arg1, IntPtr* arg1);
```

that would mean the api definition here can be removed:

https://github.com/xamarin/xamarin-macios/blob/4cfccff38ea9db5003c395bf163bd857d13adbf3/src/avfoundation.cs#L1596-L1598